orchagent/bufferorch.cpp

```diff
@@ -460,6 +466,7 @@ task_process_status BufferOrch::processBufferPool(KeyOpFieldsValuesTuple &tuple)
 
         if (SAI_NULL_OBJECT_ID != sai_object)
         {
+            clearBufferPoolWatermarkCounterIdList(sai_object);
```

Comment:
check if 'm_isBufferPoolWatermarkCounterIdListGenerated' is True before the clear?

Reply:
Sounds reasonable. Let’s check it in the clear function

Reply:
Added checking in `clearBufferPoolWatermarkCounterIdList`: only remove it if `m_isBufferPoolWatermarkCounterIdListGenerated` is true.